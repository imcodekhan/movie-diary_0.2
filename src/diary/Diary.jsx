import React, { useState, useContext } from "react";
import Tab from "./components/Tab";
import { Link } from "react-router-dom";
import userContext from "../sign/userContext";

const Diary = ({
  favMovies,
  handleTabClick,
  handleDeleteButton,
  handleNoteSave,
}) => {
  const login = useContext(userContext);
  console.log("Diary rendered");
  console.log(login);
  const [showNote, setShowNote] = useState(false);
  const [showModal, setShowModal] = useState(false);
  const [newNote, setNewNote] = useState("");
  return (
    <div className="container">
      {login ? <h3>welocme to {login.diaryname}</h3> : ""}
      <ul
        className="z-depth-3"
        style={{ margin: "10% 20%", display: "flex", flexDirection: "column" }}
      >
        {favMovies.length === 0 ? (
          <Link to="/" style={{ textDecoration: "none" }}>
            <div style={{ right: 10 }}>
              <h6
                style={{ minHeight: 100, marginLeft: "30%", marginTop: "10%" }}
              >
                Add Favourate Movies
              </h6>
              <div>
                <div
                  style={{
                    fontSize: 50,
                    fontWeight: 800,
                    height: 60,
                    width: 60,
                    border: "2px solid purple",
                    borderRadius: "50%",
                    display: "flex",
                    justifyContent: "center",
                    backgroundColor: "tomato",

                    float: "right",
                    margin: 10,
                  }}
                  onClick={() => console.log("added")}
                >
                  <h3 style={{ marginTop: 10 }}>+</h3>
                </div>
              </div>
            </div>
          </Link>
        ) : (
          favMovies.map((movie, index) => (
            <Tab
              key={movie.id}
              handleTabClick={handleTabClick}
              handleDeleteButton={handleDeleteButton}
              movie={movie}
              index={index}
              showNote={showNote}
              setShowNote={setShowNote}
              handleNoteSave={handleNoteSave}
              showModal={showModal}
              setShowModal={setShowModal}
              newNote={newNote}
              setNewNote={setNewNote}
            />
          ))
        )}
      </ul>
    </div>
  );
};

export default Diary;
