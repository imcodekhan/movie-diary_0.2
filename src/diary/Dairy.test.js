import React from "react";
import { render, cleanup, fireEvent } from "@testing-library/react";
import { createMemoryHistory } from "history";
import { Router } from "react-router-dom";
import Diary from "./Diary";

afterEach(cleanup);

const renderWithRouter = (component) => {
  const history = createMemoryHistory();
  return {
    ...render(<Router history={history}>{component}</Router>),
  };
};

const context = { user: false };
const hooks = {
  // showNote: false,
  setShowNote: jest.fn(),
  showModal: false,
  setShowModal: jest.fn(),
};

const props = {
  ...hooks,
  favMovies: [],
  handleTabClick: jest.fn(),
  handleDeleteButton: jest.fn(),
  handleNoteSave: jest.fn(),
  ...context,
};

const props1 = {
  ...props,
  favMovies: [
    {
      id: 1,
      fullTitle: "batman",
      image:
        "https://external-content.duckduckgo.com/iu/?u=https%3A%2F%2Ftse1.mm.bing.net%2Fth%3Fid%3DOIP.8CH926UPpZAdx-fpyDdBMAHaDt%26pid%3DApi&f=1",
      watched: false,
      note: "happy netflix",
    },
  ],
};

// console.log(props1.favMovies);w

it("should match non tab snapshot", () => {
  const { asFragment, getByTestId } = renderWithRouter(<Diary {...props} />);
  expect(asFragment).toMatchSnapshot();
  expect(getByTestId("diary-name")).toBeInTheDocument();
});

it("should match tab snapshot", () => {
  const { asFragment, getByText } = renderWithRouter(<Diary {...props1} />);
  expect(asFragment).toMatchSnapshot();
  expect(getByText("edit")).toBeInTheDocument();
});
