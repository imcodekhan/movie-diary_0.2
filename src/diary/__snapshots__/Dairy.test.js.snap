// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match non tab snapshot 1`] = `[Function]`;

exports[`should match tab snapshot 1`] = `[Function]`;
